pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordingrules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	abnormalLabelKey = "reason"
+)
+
+var operatorRecordingRules = []operatorrules.RecordingRule{
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "kubevirt_working_set_memory_exceeded",
+			Help: "The pod with the highest exceeded memory for each container based on the working set.",
+			ConstLabels: map[string]string{
+				abnormalLabelKey: "WorkingSetMemoryExceeded",
+			},
+		},
+		MetricType: operatormetrics.GaugeType,
+		Expr:       intstr.FromString("max by(container, namespace)(container_memory_working_set_bytes{container=~\"virt-controller|virt-api|virt-handler|virt-operator\"}  - on(pod) group_left(node) (kube_pod_container_resource_requests{container=~\"virt-controller|virt-api|virt-handler|virt-operator\",resource=\"memory\"}))"),
```

Comment:
for my own understanding, can you confirm that the namespace value isn't fixed and may vary depending on the installs?

Reply:
yes, it may vary, and @avlitman you can also filter by the namespace as in: https://github.com/kubevirt/kubevirt/blob/d4112d68b113f3d0ef0e5427c4976a86e78d1983/pkg/monitoring/rules/recordingrules/virt.go#L36

Reply:
This metric takes the pod with max memory diff per container.
Can you maybe explain why is it matter if the namespace vary? to avoid including pods that are?  

thanks!

Reply:
If the namespace value is fixed, it's way more efficient to filter by that namespace + less error-prone.

Reply:
the namespace is according to the pod with highest memory usage so I don't think it can be a const value..

Reply:
sorry I meant "a list of predictable values", e.g. `namespace=~"foo|bar"`.